handlers/http_test.go

```diff
@@ -0,0 +1,57 @@
+package handlers
+
+import (
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/storageos/discovery/cluster"
+	"github.com/storageos/discovery/store/boltdb"
+	"github.com/storageos/discovery/util/codecs"
+	"github.com/storageos/discovery/util/uuid"
+)
+
+type TestServer struct {
+	path   string
+	server *Server
+	store  *boltdb.Store
+}
+
+func setupTestServer(t *testing.T) *TestServer {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatal("could not get free port to run test server on")
+	}
+
+	port := listener.Addr().(*net.TCPAddr).Port
```

Comment:
Do we need to get a free port? I see the `nodes_test.go` tests use `httptest`, which tests the handlers directly. And I don't see the server start being called anywhere.
Maybe I'm missing something?

Reply:
No, you're right, the check is unneeded.